src/non_RL_algos/loadBalance.py

```diff
@@ -0,0 +1,163 @@
+                }
+        '''
+    Parameters:
+        nodes_list
+        sf_list
+        sfc_list
+
+    Returns:
+         schedule of the form shown above
+    """
+    schedule = defaultdict(lambda: defaultdict(lambda: defaultdict(lambda: defaultdict(float))))
+    for outer_node in nodes_list:
+        for sfc in sfc_list:
+            for sf in sf_list:
+                # all 0's list
+                uniform_prob_list = [0 for _ in range(len(nodes_list))]
+                # Uniformly distributing the schedules between all nodes
+                uniform_prob_list = normalize_scheduling_probabilities(uniform_prob_list)
+                for inner_node in nodes_list:
+                    schedule[outer_node][sfc][sf][inner_node] = uniform_prob_list.pop()
+    return schedule
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Load Balance Algorithm")
+    parser.add_argument('-i', '--iterations', required=False, default=10, dest="iterations", type=int)
+    parser.add_argument('-s', '--seed', required=False, dest="seed", type=int)
+    parser.add_argument('-n', '--network', required=True, dest='network')
+    parser.add_argument('-sf', '--service_functions', required=True, dest="service_functions")
+    parser.add_argument('-c', '--config', required=True, dest="config")
+    return parser.parse_args()
+
+
+def main():
+    # Parse arguments
+    args = parse_args()
+    if not args.seed:
+        args.seed = random.randint(1, 9999)
+    os.makedirs("logs", exist_ok=True)
+    logging.basicConfig(filename="logs/{}_{}_{}.log".format(os.path.basename(args.network),
+                                                            DATETIME, args.seed), level=logging.INFO)
+    logging.getLogger("coordsim").setLevel(logging.WARNING)
+
+    # Creating the results directory variable where the simulator result files will be written
+    network_stem = os.path.splitext(os.path.basename(args.network))[0]
+    service_function_stem = os.path.splitext(os.path.basename(args.service_functions))[0]
+    simulator_config_stem = os.path.splitext(os.path.basename(args.config))[0]
+
+    results_dir = f"{get_project_root()}/results/{network_stem}/{service_function_stem}/{simulator_config_stem}" \
+                  f"/{DATETIME}_seed{args.seed}"
+
+    # creating the simulator
+    simulator = Simulator(os.path.abspath(args.network),
+                          os.path.abspath(args.service_functions),
+                          os.path.abspath(args.config), test_mode=True, test_dir=results_dir)
+    init_state = simulator.init(args.seed)
+    log.info("Network Stats after init(): %s", init_state.network_stats)
+    nodes_list = [node['id'] for node in init_state.network.get('nodes')]
+    sf_list = list(init_state.service_functions.keys())
+    sfc_list = list(init_state.sfcs.keys())
+    ingress_nodes = get_ingress_nodes(simulator.network)
+    # we place every sf in each node of the network, so placement is calculated only once
```

Comment:
How would that work when running with the real simulator, ie, without the CLI? Which functions does the simulator call? Does the precomputation work there too?

I had an object oriented approach in mind, where the algorithm is a class and on `__init__`, it calculates placement and schedule once, stores then, and then always returns them when queried without recomputation. Not sure how that would work with the simulator. But I want to avoid recomputation in each run, which will unnecessarily increase runtime.